tree/dataframe/src/RDFActionHelpers.cxx

```diff
@@ -233,6 +233,37 @@ template class TakeHelper<float, float, std::vector<float>>;
 template class TakeHelper<double, double, std::vector<double>>;
 #endif
 
+void ValidateSnapshotOutput(const RSnapshotOptions &opts, const std::string &treeName, const std::string &fileName)
+{
+   TString fileMode = opts.fMode;
+   fileMode.ToLower();
+   if (fileMode != "update")
+      return;
+
+   // output file opened in "update" mode: must check whether output TTree is already present in file
+   std::unique_ptr<TFile> outFile{TFile::Open(fileName.c_str(), "update")};
+   if (!outFile || outFile->IsZombie())
+      throw std::invalid_argument("Snapshot: cannot open file \"" + fileName + "\" in update mode");
+
+   TObject *outTree = outFile->Get(treeName.c_str());
+   if (outTree == nullptr)
+      return;
+
+   // object called treeName is already present in the file
+   if (opts.fOverwriteIfExists) {
+      if (outTree->InheritsFrom("TTree")) {
+         static_cast<TTree *>(outTree)->Delete("all");
+      } else {
+         outFile->Delete(treeName.c_str());
+      }
+   } else {
+      const std::string msg = "Snapshot: tree \"" + treeName + "\" already present in file \"" + fileName +
+                              "\". If you want to delete the original tree and write another, please also set "
+                              "snapshot option fOverwriteIfExists to true.";
```

Comment:
Maybe `RSnapshotOptions::fOverwriteIfExists`?

Reply:
I'll change it so the CI runs one last time before merging :P